string_utils/fms_string_utils.F90

```diff
@@ -156,6 +192,86 @@ function fms_find_my_string(my_pointer, narray, string_to_find) &
 
   end function fms_find_my_string
 
+  !> \brief Converts a C-string to a pointer and then to a Fortran string
+  function cstring_fortran_conversion (cstring) result(fstring)
+    character (kind=c_char), intent(in) :: cstring (*) !< Input C-string
+    character(len=:), allocatable :: fstring    !< The fortran string returned
+    fstring = cpointer_fortran_conversion(fms_cstring2cpointer(cstring))
+  end function cstring_fortran_conversion
+
+  !> \brief Converts a C-string returned from a TYPE(C_PTR) function to
+  !! a fortran string with type character.
+  function cpointer_fortran_conversion (cstring) result(fstring)
+    type (c_ptr), intent(in) :: cstring !< Input C-pointer
+    character(len=:), allocatable :: fstring    !< The fortran string returned
+    character(len=:,kind=c_char), pointer :: string_buffer !< A temporary pointer to between C and Fortran
+    integer(c_size_t) :: length !< The string length
+
+    length = c_strlen(cstring)
+    allocate (character(len=length, kind=c_char) :: string_buffer)
+    block
+      character(len=length,kind=c_char), pointer :: s
+      call c_f_pointer(cstring,s)  ! Recovers a view of the C string
+      string_buffer = s                   ! Copies the string contents
+    end block
+
+    allocate(character(len=length) :: fstring) !> Set the length of fstring
+    fstring = string_buffer
+    deallocate(string_buffer)
+  end function cpointer_fortran_conversion
+
+  !> @brief Converts an integer to a string
+  !!
+  !> This has been updated from the fms_io function.
+  function string_from_integer(i) result (res)
```

Comment:
Needs a `\result`.  I don't think it needs this updated from fms_io comment.  That will live there beyond our lives.